### PR TITLE
Added error when splatting inside hvcat syntax

### PR DESCRIPTION
**Pull request by BioTurboNick**
As raised in issue #38844, splatting inside hvcat syntax produces unexpected output. This PR detects this case and emits an error.

### CODE REVIEW

File: src/julia-syntax.scm [hunk 1]
```diff
@@ -2424,6 +2424,10 @@
                                        (cdr x)
                                        (list x)))
                                  a)))
+                  (if (any (lambda (x)
+                             (and (pair? x) (pair? (car x)) (eq? (caar x) '...)))
```
Comment (simeonschaub) [inline src/julia-syntax.scm:2428]: Doesn't this only check the first item in each row? `(pair? (car x)) (eq? (caar x) '...))` could also be replaced by `(vararg? (car x))`.

Comment (BioTurboNick) [reply src/julia-syntax.scm:2428]: Of course, you're right. Wasn't aware of the vararg function, I'll try it.

File: src/julia-syntax.scm [hunk 2]
```diff
@@ -2424,6 +2424,12 @@
                                        (cdr x)
                                        (list x)))
                                  a)))
+                  (if (any (lambda (x)
+                             (and (pair? x) (any (lambda (y)
+                                                   (and (pair? y) (eq? (car y) '...)))
+                                            x)))
+                           rows)
```
Comment (simeonschaub) [inline src/julia-syntax.scm:2431]:
```suggestion
                  (if (any (lambda (x) (any vararg? x)) rows)
```

**Review by simeonschaub — COMMENTED** [review]
Thanks for doing this! I just have one style suggestion. Would you mind also adding a test for this?

**Comment by BioTurboNick** [comment]
Just added tests. Thanks for your comments!

**Review by simeonschaub — APPROVED** [review]
LGTM, thanks!

**Comment by simeonschaub** [comment]
@JeffBezanson Could you take a quick look at this?

**Comment by BioTurboNick** [comment]
Linking to #39249 which would render this PR unnecessary.

**Comment by BioTurboNick** [comment]
Closing because #39249 was merged

**Pull request closed without merging**